### PR TITLE
Add workflow_upload command

### PR DESCRIPTION
**Pull request by mvdbeek**
`planemo workflow_upload` will update or create a github repository with the contents of a directory that contains a workflow.
Every upload will create a new release. Workflow versions are taken from the workflow files.

Also moves from `hub` to `gh` and makes `planemo ci_find_repos` return directories with workflows.

TODO:
- [x] Check if version already exists / handle empty diff ?
- [ ] `--force` that will override existing version
- [x] Handle deleted files (remove everything in work dir, then copy ?)

### CODE REVIEW

File: planemo/ci.py [hunk 1]
```diff
@@ -29,21 +32,16 @@ def filter_paths(ctx, raw_paths, path_type="repo", **kwds):
...
+                diff_path = metadata_file_in_path(diff_dir)
+                if diff_path:
+                    diff_paths.add(diff_path)
+                    break
         else:
             diff_paths = diff_files
 
     unique_paths = set(os.path.relpath(p, cwd) for p in raw_paths)
     if diff_paths is not None:
-        new_unique_paths = []
```
Comment (mvdbeek) [inline planemo/ci.py:42]: I don't think I have understood the purpose here. Was this meant to filter our paths in `diff_paths` that are not within `raw_paths` ?

**Comment by mvdbeek** [comment]
I think this is ready for review!

**Review by jmchilton — APPROVED** [review]

**Comment by jmchilton** [comment]
Simply amazing!

**Comment by bgruening** [comment]
Wonderful!

File: planemo/ci.py (hunk 1 above)
Comment (nsoranzo) [reply planemo/ci.py:42]: This was (and should remain) the intersection of `unique_paths` (i.e. the tool or repo specified by path by the user) and `diff_paths` (the list of files or directories changed in Git).
Sorry I missed this at the time, should be fixed by https://github.com/galaxyproject/planemo/pull/1127 .

**Pull request merged**